### ISSUE TITLE
build error with rustc 1.2.0-nightly

### ISSUE THREAD

**Issue by Arnold1** [sivadeilra/racc#4]
Hi, 

I use rustc 1.2.0-nightly (fbb13543f 2015-06-11).

there are several build errors in src/output.rs and src/reader.rs, e.g.:
```
src/reader.rs:623:44: 623:48 help: add #![feature(rustc_private)] to the crate attributes to enable
src/reader.rs:626:24: 626:30 error: use of unstable library feature 'rustc_private'
```

Do you plan to fix it?

**Comment by brandonchinn178** [comment]
Yea, I still got `rustc_private` compile time errors, but I didn't know how to deal with it

**Comment by Arnold1** [comment]
the rustc_private issue is fixed, but more important is the following issue, do you know how to fix it?
```
src/bin/testg.rs:26:1: 26:8 error: macro undefined: 'grammar!'
src/bin/testg.rs:26 grammar! {
```

**Comment by brandonchinn178** [comment]
1. Is `#[macro_export]` above the `grammar` macro_rule?
2. Is `#[macro_use]` above the export?
3. Are you importing the `racc` package above all other modules?

**Comment by Arnold1** [comment]
i get the error when i build the racc project: cargo build...

**Comment by Arnold1** [comment]
can you comment?

**Comment by brandonchinn178** [comment]
I don't know, I think it has something to do with how @sivadeilra declared the `macro_rule`

**Comment by Arnold1** [comment]
there are the following decl in the testg.rs file:
```
#![feature(plugin)]

#![allow(non_upper_case_globals)]
#![allow(dead_code)]
#![allow(non_camel_case_types)]
#![allow(unused_variables)]
#![allow(unstable)]

#[plugin]
#[macro_use]
extern crate log;
```

**Comment by geraldstanje** [comment]
hi, i fixed the build for rust nightly. please merge.

**Comment by sivadeilra** [comment]
Merged.  Thanks!

**Comment by geraldstanje** [comment]
does it work as expected? what is missing in the project? can you update the readme?

**Comment by sivadeilra** [comment]
Sorry, I merged it without testing it. I've been out of the loop for a
while, now.

**Comment by geraldstanje** [comment]
do you plan to jump into the loop again?

**Comment by sivadeilra** [comment]
Yes.  I'm out of town for most of July but intend to pick up development when I return.

-----Original Message-----

**Comment by geraldstanje** [comment]
please merge my update.
